File: _CodeTopics/LeetCode_contest/weekly/weekly2020/187/187_1.py
class Solution(object):
    def destCity(self, paths):
        """
        :type paths: List[List[str]]
        :rtype: str
        """
        
        resdic = {}
        for path in paths:    
            resdic[path[0]] = 0
            if resdic.has_key(path[1]) is False:
                resdic[path[1]] = 1
        
        for k, v in resdic.items():
            if v == 1:
                return k
            
"""
103 / 103 个通过测试用例
	状态：通过
执行用时：24 ms
内存消耗：12.7 MB
"""

File: _CodeTopics/LeetCode/1401-1600/001436--weekly187_1/tran/187_1.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {

        // 第 187 周周赛第一题
        // https://leetcode-cn.com/submissions/detail/67765940/

        unordered_map<string, int> resdic = {};
        for (auto path : paths) {
            resdic[path[0]] = 0;
            if (resdic.find(path[1]) == resdic.end()) {
                resdic[path[1]] = 1;
            }
        }
        
        string res = "";
        for (auto kvpair : resdic) {
            if (kvpair.second == 1) {
                res = kvpair.first;
                break;
            }
        }
        return res;
    }
};

/*
https://leetcode-cn.com/submissions/detail/224859826/

103 / 103 个通过测试用例
状态：通过
执行用时: 12 ms
内存消耗: 11.1 MB

执行用时：12 ms, 在所有 C++ 提交中击败了78.83%的用户
内存消耗：11.1 MB, 在所有 C++ 提交中击败了34.85%的用户
*/
